setup/providers/azure.md

```diff
@@ -7,8 +7,77 @@ sidebar:
 
 {% include toc %}
 
+In [Azure](https://azure.microsoft.com/), an [__Account__](/setup/providers/#accounts) maps to a
+credential able to authenticate against a given [Azure subscription](https://azure.microsoft.com/free/).
+
 ## Prerequisites
 
+You need a [Service Principal](https://docs.microsoft.com/cli/azure/create-an-azure-service-principal-azure-cli) to authenticate with Azure and a [Key Vault](https://azure.microsoft.com/services/key-vault/) to store a default username/password for deployed [VM Scale Sets](https://docs.microsoft.com/azure/virtual-machine-scale-sets/virtual-machine-scale-sets-overview). The next steps assume the use of the [Azure CLI 2.0](https://docs.microsoft.com/cli/azure/install-azure-cli). The example commands will set environment variables along the way for use when creating an account in the final stage. You can check that you have `az` installed by running:
+
+```bash
+az --version
+```
+
+First, log in and set your subscription:
+
+```bash
+az login
+az account list
+SUBSCRIPTION_ID=<Insert Subscription ID>
+az account set --subscription $SUBSCRIPTION_ID
+```
+
+Next, create a Service Principal and set environment variables based on the output:
+
+```bash
+az ad sp create-for-rbac --name "Spinnaker"
+APP_ID=<Insert App Id>
+TENANT_ID=<Insert Tenant Id>
+```
+
+> NOTE: You will need the App Key (also called password) when creating an account, but you will be prompted on standard input for that since it is sensitive data.
```

Comment:
Add a note that the name "Spinnaker" used for the creation of the service principal must be unique in the Azure subscription.